library/x509_crt.c

```diff
@@ -614,11 +682,13 @@ static int x509_get_ext_key_usage( unsigned char **p,
  *      nameAssigner            [0]     DirectoryString OPTIONAL,
  *      partyName               [1]     DirectoryString }
  *
- * NOTE: we only parse and use dNSName at this point.
+ * NOTE: we only parse and use dNSName at this point,
+ * and otherName of type HwModuleName, as defined in RFC 4108.
  */
 static int x509_get_subject_alt_name( unsigned char **p,
                                       const unsigned char *end,
-                                      mbedtls_x509_sequence *subject_alt_name )
+                                      mbedtls_x509_sequence *subject_alt_name,
```

Comment:
Similar to above, I think the naming is not optimal here. Now that we distinguish more than one kind of subject alternative name, with `subject_alt_name` only holding the `dNSName` entries, and `hardware_module_name` the `HwModuleName` entries, the parameter names should reflect that.

However, again this is not a blocker because the function is `static`, and is also largely rewritten in #2478 anyway.

Reply:
The subject_alt_name should hold more than just `dnsName` in the future, such as `ipAddress.`
However, the `HardwareModuleName` is a different structure, as it is actually  a sequence of an oid and an Octet String, while `dnsName` and `iPAddress` are strings. We could think of a way to combine the two